### PR TITLE
[SG-73] Make Organization Info a reusable component

### PR DESCRIPTION
**Pull request by differsthecat**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Create new organization info component to be used by the Trial Initiation stepper and to replace the current organization info code.

Depends on some changes in https://github.com/bitwarden/clients/pull/2946

## Code changes

- **apps/web/src/app/modules/organizations/create:** Add new `create` module to the organizations module to house the new Organization Information component.
- **apps/web/src/app/settings/organization-plans.component.html:**  Replace old form items with `<app-org-info>`
- **apps/web/src/app/settings/organization-plans.component.ts:** Get new form values when they are updated.

**NOTE** I currently have this component contain it's own formgroup, but I can modify this to pass in one from a parent component depending on how billing is handled.

## Screenshots

https://user-images.githubusercontent.com/8926729/176513112-123b155f-dbe8-4884-801c-715d328820d7.mov

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

File: apps/web/src/app/settings/organization-plans.component.ts [hunk 1]
```diff
@@ -167,6 +168,12 @@ export class OrganizationPlansComponent implements OnInit {
     return plan.seatPrice * Math.abs(this.additionalSeats || 0);
   }
 
+  orgInfoValueChange(form: FormGroup) {
```
Comment (gbubemismith) [inline apps/web/src/app/settings/organization-plans.component.ts:171]: I am wondering if we have some time to also convert the organization plans to use reactive forms as well. That way, we can utilize sub forms in the parent form

File: apps/web/src/app/modules/organizations/create/organization-information.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+@Component({
+  selector: "app-org-info",
+  templateUrl: "organization-information.component.html",
+})
+export class OrganizationInformationComponent implements OnInit {
+  @Input() nameOnly = false;
+  @Output() valueChange = new EventEmitter<FormGroup>();
+
+  formData = this.formBuilder.group({
```
Comment (gbubemismith) [inline apps/web/src/app/modules/organizations/create/organization-information.component.ts:12]: I think we can have one form group with all the controls

**Review by gbubemismith — COMMENTED** [review]
Looks good but I added some few suggestions

File: apps/web/src/app/settings/organization-plans.component.ts (hunk 1 above)
Comment (differsthecat) [reply apps/web/src/app/settings/organization-plans.component.ts]: Done!

File: apps/web/src/app/modules/organizations/create/organization-information.component.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<form #form [formGroup]="formData" *ngIf="nameOnly">
```
[the remark this answers is not in the document]
Comment (gbubemismith) [reply apps/web/src/app/modules/organizations/create/organization-information.component.html]: yeah I checked and it's fine

**Review by gbubemismith — APPROVED** [review]
This looks good

**Pull request merged**